src/frontend/planner_test/tests/testdata/share.yaml

```diff
@@ -141,3 +141,45 @@
                             └─StreamProject { exprs: [auction, date_time, _row_id] }
                               └─StreamRowIdGen { row_id_index: 7 }
                                 └─StreamSource { source: "bid", columns: ["auction", "bidder", "price", "channel", "url", "date_time", "extra", "_row_id"] }
+- sql: |
+    set rw_enable_share_plan=true;
+    create table t(a int, b int);
+    with cte as (select count(*) from t) select * from cte union all select * from cte;
+  stream_plan: |
+    StreamMaterialize { columns: [count, 0:Int32(hidden)], pk_columns: [0:Int32] }
+    └─StreamUnion { all: true }
+      ├─StreamExchange { dist: HashShard(0:Int32) }
+      | └─StreamProject { exprs: [sum0(count), 0:Int32] }
+      |   └─StreamShare { id = 213 }
+      |     └─StreamProject { exprs: [sum0(count)] }
+      |       └─StreamGlobalSimpleAgg { aggs: [count, sum0(count)] }
+      |         └─StreamExchange { dist: Single }
+      |           └─StreamStatelessLocalSimpleAgg { aggs: [count, count] }
+      |             └─StreamTableScan { table: t, columns: [t._row_id], pk: [t._row_id], dist: UpstreamHashShard(t._row_id) }
+      └─StreamExchange { dist: HashShard(1:Int32) }
+        └─StreamProject { exprs: [sum0(count), 1:Int32] }
+          └─StreamShare { id = 213 }
+            └─StreamProject { exprs: [sum0(count)] }
+              └─StreamGlobalSimpleAgg { aggs: [count, sum0(count)] }
+                └─StreamExchange { dist: Single }
+                  └─StreamStatelessLocalSimpleAgg { aggs: [count, count] }
+                    └─StreamTableScan { table: t, columns: [t._row_id], pk: [t._row_id], dist: UpstreamHashShard(t._row_id) }
+- sql: |
+    set rw_enable_share_plan=false;
+    create table t(a int, b int);
+    with cte as (select count(*) from t) select * from cte union all select * from cte;
+  stream_plan: |
```

Comment:
When `rw_enable_share_plan = false`, will it create 2 source executors? That might be a bug for self-join cases. #4618

Reply:
Good catch! I miss this point.